### PR TITLE
Add record, intent, stream and export to elasticsearch for signal

### LINKED ISSUE #11295: Create SignalRecord and Signalntent
## Description
Introduce new `Signal` records, with intents: `Broadcast`, `Broadcasted`
- The signal record value should consist of:
  - `signalName  `
  - optionally `variables`
  - Like any other records, these should be exported. Please follow our [developer handbook entry on adding new records](https://github.com/camunda/zeebe/blob/main/docs/developer_handbook.md#how-to-create-a-new-record)

**Part of**
[Broadcast a signal using the gRPC API #11196](https://github.com/camunda/zeebe/issues/11196)

### PR DESCRIPTION
**Pull request by skayliu**
## Description

Add record, intent, stream and export to elasticsearch for signal.

## Related issues

closes #11295 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by skayliu** [comment]
Hi, @korthout, Please take a look at this feat. Thank you. :heart:

File: test-util/src/main/java/io/camunda/zeebe/test/util/record/CompactRecordLogger.java [hunk 1]
```diff
@@ -87,6 +88,7 @@ public class CompactRecordLogger {
           entry("EVENT", "EVNT"),
           entry("DECISION_REQUIREMENTS", "DRG"),
           entry("EVALUATION", "EVAL"),
+          entry("SIGNAL", "SIG"),
           entry("SIGNAL_SUBSCRIPTION", "SIG_SUB"));
```
Comment (korthout) [inline test-util/src/main/java/io/camunda/zeebe/test/util/record/CompactRecordLogger.java:92]: ❌ Please swap these. The order matters. In the current order, records of type `SIGNAL_SUBSCRIPTION` would be abbreviated to `SIG_SUBSCRIPTION`  instead `SIG_SUB`.

File: test-util/src/main/java/io/camunda/zeebe/test/util/record/CompactRecordLogger.java [hunk 2]
```diff
@@ -718,6 +721,16 @@ private String summarizeDecisionInformation(final String decisionId, final long
...
   }
 
+  private String summarizeSignal(final Record<?> record) {
+    final var value = (SignalRecordValue) record.getValue();
+
+    final var result = new StringBuilder().append("\"").append(value.getSignalName()).append("\"");
+
+    result.append(summarizeVariables(value.getVariables()));
+
+    return result.toString();
```
Comment (korthout) [inline test-util/src/main/java/io/camunda/zeebe/test/util/record/CompactRecordLogger.java:731]: 🔧 There's no need for the local variable result

```java
return new StringBuilder()
    .append("\"")
    .append(value.getSignalName())
    .append("\"")
    .append(summarizeVariables(value.getVariables()))
    .toString();
```

**Review by korthout — CHANGES_REQUESTED** [review]
Nicely done @skayliu 👏 

🔧 We want `Signal:Broadcast` commands to be sent from the client to the gateway, and from the gateway to the broker. To support the broker receiving this command from the gateway, we'll need to add the `Signal` record to the `records_by_type` mapping in `CommandApiRequestReader`. We could add it later, but it's easy to miss and complicates efforts once you start work on the gateway. So, I suggest making that change now.

❌ There's one small change required before merging this pull request (i.e. the order of the compact record logger's entries). Please have a look at my comment below.

👍 Awesome job on splitting up the commits. The changes were easy to follow.

**Comment by skayliu** [comment]
@korthout Done, Please check this out. Thank you.

**Review by korthout — APPROVED** [review]
Awesome 🚀 

LGTM 💯 

🙇 Sorry for the slow review @skayliu, let's get this merged

bors merge

**Comment by korthout** [comment]
bors retry

**Comment by korthout** [comment]
Sorry about that @skayliu, we're experiencing some difficulties with out Github Actions Self-Hosted Runners. I'll retry when I think it should be resolved

**Comment by skayliu** [comment]
> Sorry about that @skayliu, we're experiencing some difficulties with out Github Actions Self-Hosted Runners. I'll retry when I think it should be resolved

Take your time& there is no rush.

**Pull request merged**